File: CompCode/USACO_Bronze/LostCow/LostCow.cpp
/*
ID: florinteo
PROG: LostCow
LANG: Java             
*/
/*
DEF: https://usaco.org/index.php?page=viewproblem2&cpid=735
*/
#include <iostream>
#include <fstream>
#include <sstream>

using namespace std;

int main() {
    #pragma region load data
    ifstream cin("lostcow.in");
    int x, y;
    cin >> x >> y;
    #pragma endregion load data

    #pragma region run algorithm
    int totalDistance = 0;
    int step = 1;
    int crtX = x;
    int newX = x + step;
    while((y - crtX) * (y - newX) > 0) {
        totalDistance += abs(newX - crtX);
        crtX = newX;
        step *= -2;
        newX = x + step;
    }
    totalDistance += abs(y - crtX);
    #pragma endregion run algorithm

    #pragma region write output
    ofstream cout("lostcow.out");
    cout << totalDistance << endl;
    #pragma endregion write output

    #pragma region cleanup
    #pragma endregion cleanup

    return 0;
}

File: CompCode/USACO_Bronze/LostCow/LostCow.java
/*
ID: florinteo
PROG: LostCow
LANG: Java             
*/
/*
DEF: https://usaco.org/index.php?page=viewproblem2&cpid=735
*/

import java.io.BufferedReader;
import java.io.FileReader;
import java.io.IOException;
import java.io.PrintWriter;
import java.util.StringTokenizer;

public class LostCow {
    public static void main(String[] args) throws IOException {
        // #region load data
        BufferedReader br = new BufferedReader(new FileReader("USACO_Bronze/LostCow/lostcow.in"));
        StringTokenizer st = new StringTokenizer(br.readLine());
        int x = Integer.parseInt(st.nextToken());
        int y = Integer.parseInt(st.nextToken());
        br.close();
        // #endregion load data

        // #region algorithm
        int totalDistance = 0;
        int step = 1;
        int crtX = x;
        int newX = x + step;
        while((y - crtX) * (y - newX) > 0) {
            totalDistance += Math.abs(newX - crtX);
            crtX = newX;
            step *= -2;
            newX = x + step;
        }
        totalDistance += Math.abs(y - crtX);
        // #endregion algorithm

        // #region write output
        PrintWriter pw = new PrintWriter("USACO_Bronze/LostCow/lostcow.out");
        pw.println(totalDistance);
        pw.flush();
        pw.close();
        // #endregion write output
    }
}
